src/ipc/handler.c

```diff
@@ -1246,8 +1261,14 @@ int ipc_queue_host_message(struct ipc *ipc, uint32_t header, void *tx_data,
 	msg->tx_size = tx_bytes;
 
 	/* copy mailbox data to message */
-	if (tx_bytes > 0 && tx_bytes < SOF_IPC_MSG_MAX_SIZE)
-		rmemcpy(msg->tx_data, tx_data, tx_bytes);
+	if (tx_bytes > 0 && tx_bytes < SOF_IPC_MSG_MAX_SIZE) {
+		ret = memcpy_s(msg->tx_data, msg->tx_size, tx_data, tx_bytes);
+		if (ret != 0) {
+			trace_ipc_error("ipc_queue_host_message()"
+					"could not coppy data");
```

Comment:
Could do add copy size in error message? It is useful for debug.

Reply:
I will in separate commit as agreed with @lgirdwood